src/core/server/config/__tests__/env.test.ts

```diff
@@ -29,76 +29,82 @@ jest.mock('path', () => ({
   },
 }));
 
+const mockPackage = new Proxy({ raw: {} as any }, { get: (obj, prop) => obj.raw[prop] });
+jest.mock('../../../../utils/package_json', () => ({ pkg: mockPackage }));
+
 import { Env } from '../env';
-import { getEnvOptions } from './__mocks__/env';
-
-test('correctly creates default environment with empty options.', () => {
-  const envOptions = getEnvOptions();
-  const defaultEnv = Env.createDefault(envOptions);
-
-  expect(defaultEnv.homeDir).toEqual('/test/cwd');
-  expect(defaultEnv.configDir).toEqual('/test/cwd/config');
-  expect(defaultEnv.corePluginsDir).toEqual('/test/cwd/core_plugins');
-  expect(defaultEnv.binDir).toEqual('/test/cwd/bin');
-  expect(defaultEnv.logDir).toEqual('/test/cwd/log');
-  expect(defaultEnv.staticFilesDir).toEqual('/test/cwd/ui');
-
-  expect(defaultEnv.getConfigFile()).toEqual('/test/cwd/config/kibana.yml');
-  expect(defaultEnv.getLegacyKbnServer()).toBeUndefined();
-  expect(defaultEnv.getMode()).toEqual(envOptions.mode);
-  expect(defaultEnv.getPackageInfo()).toEqual(envOptions.packageInfo);
+
+test('correctly creates default environment in dev mode.', () => {
+  mockPackage.raw = {
+    branch: 'some-branch',
+    version: 'some-version',
+  };
+
+  const defaultEnv = Env.createDefault({
+    cliArgs: { dev: true, someArg: 1, someOtherArg: '2' },
+    configs: ['/test/cwd/config/kibana.yml'],
+    isDevClusterMaster: true,
+  });
+
+  expect(Object.entries(defaultEnv)).toMatchSnapshot('env properties');
```

Comment:
Why can't this just check `expect(defaultEnv).toMatchSnapshot()`? Doesn't `Object.entries()` make this sensitive to property ordering? I certainly think it makes the snapshots harder to grok. If it has to do with hidden properties or something maybe `expect({ ...defaultEnv }).toMatchSnapshot()` would be a slightly better choice?

Reply:
> Why can't this just check expect(defaultEnv).toMatchSnapshot()?

Hmm, we can, I have no idea why I have these `Object.entries` :see_no_evil: